include/openssl/opensslconf.h.in

```diff
@@ -97,7 +99,13 @@ extern "C" {
 # define OPENSSL_API_COMPAT OPENSSL_MIN_API
 #endif
 
-#if OPENSSL_API_COMPAT < 0x10200000L
+/*
+ * Do not deprecate things to be deprecated in version 1.2.0 before the
+ * OpenSSL version number matches.
+ */
+#if OPENSSL_VERSION_NUMBER < 0x10200000L
+# define DEPRECATEDIN_1_2_0(f)   f;
+#elif OPENSSL_API_COMPAT < 0x10200000L
```

Comment:
This of course makes the rest of this dead code. Should we be leaving it there? And of course as soon as we hit 1.2.0, the first bit becomes dead instead. An alternative is to have something like `#error OpenSSL version number changed` in the `else` bit, and then revise this code when we actually get to 1.2.0

Reply:
The point is that with this code, we don't need to bother further, everything will work out automatically when we hit 1.2.0, and we can simply choose to change this when we feel like, or not at all.  I guess I took the Pooh route...  you know, "oh bother" ;-)

Reply:
I noticed this but couldn't think of a better alternative.
Arithmetic expressions might work, but I'm not seeing how.  Unfortunately.

Reply:
I prefer not to have dead code hanging around. I'd rather that we got an explicit error and a prompt to update things when the version number hits 1.2.0. But this is just my suggestion. Feel free to take @paulidale's approval.

Reply:
> Feel free to take @paulidale's approval.

Okie :-)